tests/unit/browser/webengine/test_webenginetab.py

```diff
@@ -0,0 +1,109 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2018 Florian Bruhin (The Compiler) <mail@qutebrowser.org>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Test webenginetab."""
+
+from unittest import mock
+import logging
+
+from PyQt5.QtCore import QObject
+import pytest
+QtWebEngineWidgets = pytest.importorskip("PyQt5.QtWebEngineWidgets")
+QWebEnginePage = QtWebEngineWidgets.QWebEnginePage
+QWebEngineScriptCollection = QtWebEngineWidgets.QWebEngineScriptCollection
+
+from qutebrowser.browser.webengine.webenginetab import _WebEngineScripts
+from qutebrowser.browser.greasemonkey import GreasemonkeyScript
+
+pytestmark = pytest.mark.usefixtures('greasemonkey_manager')
+
+
+class TestWebengineScripts:
+    """Test the _WebEngineScripts utility class."""
+
+    class FakeWidget(QObject):
+        """Fake widget for _WebengineScripts to use."""
```

Comment:
Hmm, makes me wonder if it's really worth it to create a fake widget like that - @toofar, what do you think about just using a real `QWebEngineView` there?

Reply:
Maybe we also should use `FakeWebTab` which is in `tests/helpers/stubs.py` already instead? Sure, you only need the scripts part here, but that way any other tests needing a fake scripts object could use the same fixture.

Reply:
So use the  `webengine_tab` fixture and add scripts to it and then check `page().scripts()` and caplog for the expected results? It seems rather heavyweight for testing the logic of just one function. I have been trying to avoid slow tests so we don't go from having slow end2end tests to slow "unit" tests.

Reply:
Oh, I just discovered what took a long time was usually showing the widget, which isn't needed in many tests such as those - with f63eb8ea15335dd34ee5403332dec39dacdf0f7b, it's down to ~0.8s on my machine for all tests (instead of ~0.6s with the mocking).